capstone/src/main/java/com/google/sps/servlets/clubs/ClubServlet.java

```diff
@@ -72,10 +72,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
               logoKey,
               calendar,
               labels,
+              isExclusive,
               creationTime);
       Gson gson = new Gson();
       JsonElement jsonElement = gson.toJsonTree(club);
+      boolean isOfficer = officers.contains(userEmail);
+      boolean viewable = members.contains(userEmail) || !isExclusive;
       jsonElement.getAsJsonObject().addProperty("isOfficer", isOfficer);
+      jsonElement.getAsJsonObject().addProperty("viewable", viewable);
```

Comment:
It looks like you're trusting the client side to just not display clubs that are marked not viewable - ideally the server should refuse to provide the data at all, since then you could just peek at the GET'ed data and find all the club info anyway.

Reply:
That makes sense, thanks!